packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -156,6 +156,102 @@ contract OptimismPortal_Test is CommonTest {
         assertEq(address(optimismPortal).balance, _value);
     }
 
+    /// @dev Tests that `depositTransaction` succeeds when msg.sender == tx.origin and non-custom gas is used.
+    function testFuzz_depositTransaction_senderIsOrigin_succeeds(
+        address _to,
+        uint256 _mint,
+        uint256 _value,
+        uint64 _gasLimit,
+        bool _isCreation,
+        bytes memory _data
+    )
+        external
+    {
+        if (_isCreation) {
+            vm.assume(_to == address(0));
+        }
+        vm.assume(_data.length <= 120_000);
+        vm.assume(_gasLimit >= optimismPortal.minimumGasLimit(uint64(_data.length)));
+        ResourceMetering.ResourceConfig memory rcfg = systemConfig.resourceConfig();
+        vm.assume(int256(uint256(_gasLimit)) <= int256(uint256(rcfg.maxResourceLimit)));
+
```

Comment:
It could be good to use `bound` rather than `vm.assume` here because it will result in a lot of dropped fuzz runs. If you hit too many dropped runs then it errors the test run and also can increase the time of the test